pylint/utils.py

```diff
@@ -225,52 +225,7 @@ def _checker_messages(self, checker):
 
     def disable(self, msgid, scope='package', line=None, ignore_unknown=False):
         """don't output message of the given id"""
-        assert scope in ('package', 'module')
-        # handle disable=all by disabling all categories
-        if msgid == 'all':
-            for message_id in MSG_TYPES:
-                self.disable(message_id, scope, line)
-            return
-        # msgid is a category?
-        catid = category_id(msgid)
-        if catid is not None:
-            for _msgid in self.msgs_store._msgs_by_category.get(catid):
-                self.disable(_msgid, scope, line)
-            return
-        # msgid is a checker name?
-        if msgid.lower() in self._checkers:
-            msgs_store = self.msgs_store
-            for checker in self._checkers[msgid.lower()]:
-                for _msgid in checker.msgs:
-                    if _msgid in msgs_store._alternative_names:
-                        self.disable(_msgid, scope, line)
-            return
-        # msgid is report id?
-        if msgid.lower().startswith('rp'):
-            self.disable_report(msgid)
-            return
-
-        try:
-            # msgid is a symbolic or numeric msgid.
-            msg = self.msgs_store.check_message_id(msgid)
-        except UnknownMessageError:
-            if ignore_unknown:
-                return
-            raise
-
-        if scope == 'module':
-            self.file_state.set_msg_status(msg, line, False)
-            if msg.symbol != 'locally-disabled':
-                self.add_message('locally-disabled', line=line,
-                                 args=(msg.symbol, msg.msgid))
-
-        else:
-            msgs = self._msgs_state
-            msgs[msg.msgid] = False
-            # sync configuration object
-            self.config.disable = [self._message_symbol(mid)
-                                   for mid, val in sorted(six.iteritems(msgs))
-                                   if not val]
+        self._set_msg_status(msgid, False, scope, line, ignore_unknown)
```

Comment:
Pass these as keyword arguments, so that we won't pass False as an argument and know what it should mean.